test/test_parser.rb

```diff
@@ -1856,6 +1856,7 @@ def test_def
       %q{def foo; end},
       %q{~~~ keyword
         |    ~~~ name
+        |! assignment
```

Comment:
@iliabylich I'm glad you suggested we introduce this notation, didn't think I'd use it so soon 🙇‍♂️